onmt/dynamic/vocab.py

```diff
@@ -0,0 +1,87 @@
+"""Module for build dynamic fields."""
+from collections import Counter, defaultdict
+import os
+import torch
+from onmt.utils.logging import logger
+from onmt.inputters.inputter import get_fields, _load_vocab, \
+    _build_fields_vocab
+
+
+def _no_tokenize(line):
+    """Return `line` as is.
+
+    Function used in _get_dynamic_fields, made first-level to be picklable.
+    """
+    return line
+
+
+def _get_dynamic_fields():
+    # TODO: support for features & other opts
+    src_nfeats = 0
+    tgt_nfeats = 0
+    fields = get_fields('text', src_nfeats, tgt_nfeats)
+
+    src_base_field = fields['src'].base_field
+    tgt_base_field = fields['tgt'].base_field
+    src_base_field.tokenize = _no_tokenize
+    tgt_base_field.tokenize = _no_tokenize
+    # Do not support this in dynamic for now
+    del fields['corpus_id']
```

Comment:
Why?

Reply:
I haven't look deep for 'corpus_id', but since it's a Field and require Vocab, it needed some specific handling for it. Therefore, I haven't integrated it into the code. And at my very first glance, if I am not missing something, this field is used to adding noise to source/target. I think we can do it in the transform class which will be smarter and easy to understand.

Reply:
BART noise is added to the dynamic transform. As our dynamic support corpus level transform configuration, so this 'corpus_id' field which intends to selectively inject noise to some corpus is redundant in our dynamic regime and can be ignored safely.